### PR TITLE
e2e test now tests top-level run()

### PR DESCRIPTION
**Pull request by bittrance**
Previously, the e2e tests invoked each `*Command` method directly and there was no coverage of the actual cli invocation. Goland appears not to have a good method to invoke the binary being built, so we cannot easily exec it, but we can at least invoke the top-level `run()` method, which takes a `[]string` of cli args to parse. This unearthed an invalid argument ordering introduced as part of changes in #62 which would have made the provider unusable.

This PR also prepares the ground for running these tests against the container to verify that dependency changes don't break the tool.

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -25,3 +31,148 @@ func TestCIRequirements(t *testing.T) {
...
+		providerType:  "github",
+		username:      "gitops-promotion",
+		password:      os.Getenv("GITHUB_TOKEN"),
+		url:           os.Getenv("GITHUB_URL"),
+		defaultBranch: "main",
+	},
+}
+
+func runCommand(command string, args ...string) (string, error) {
+	var cmdline = []string{"ze-binary", command}
```
Comment (simongottschlag) [inline main_test.go:61]: Helper functions usually start with `test` - `testRunCommand`. They also usually take `t *testing.T` and starts with `t.Helper()`.

File: main_test.go [hunk 2]
```diff
@@ -25,3 +31,148 @@ func TestCIRequirements(t *testing.T) {
...
+			require.NoError(t, err)
+
+			require.Equal(t, "created promotions pull request", newCommandMsgDev)
+
+			path = testCloneRepositoryAndValidateTag(t, p.url, p.username, p.password, p.defaultBranch, group, "dev", app, tag)
+
+			repoDev := testGetRepository(t, path)
+			revDev := testGetRepositoryHeadRevision(t, repoDev)
+
+			testSetStatus(t, ctx, providerType, revDev, group, "dev", p.url, p.password, true)
```
Comment (simongottschlag) [inline main_test.go:107]: For readability, moving this to a new helper `testFirstEnvironment()` would be possible and then having the same helper function for QA and PROD like `testEnvironment()`. Naming is hard and these names are just bad examples.

**Review by simongottschlag — APPROVED** [review]
Love it! 😊 The description says `Goland` when I think it should say `Golang`.

File: main_test.go (hunk 1 above)
Comment (bittrance) [reply main_test.go:61]: This is not a helper, tho. It actually wraps the execution of the code under test. I think that if we call `Helper()` here, we will "exclude" the entire test, I think?

Comment (simongottschlag) [reply main_test.go:61]: My understanding is that a function in a test file that is only used with tests is a test helper function - but that may be a personal preference more than anything and not something I see as a requirement. I usually name these functions `testFunction` with the `test`-prefix.

The reason `t.Helper()` exists is to not print information about that specific function when something fails and you know best if that's needed here. Linters usually remind on this.

You choose, feel free to set as resolved.

File: main_test.go (hunk 2 above)
Comment (simongottschlag) [reply main_test.go:107]: I still think this would be nice, but not a must have.

**Review by simongottschlag — DISMISSED** [review]
Looking good!

File: main_test.go (hunk 1 above)
Comment (bittrance) [reply main_test.go]: Fixed.

**Review by simongottschlag — APPROVED** [review]
LGTM 😊🥇

**Pull request merged**